src/nuklear.h

```diff
@@ -2666,7 +2689,8 @@ NK_API void nk_group_set_scroll(struct nk_context*, const char *id, nk_uint x_of
 ///
 /// Returns `true(1)` if visible and fillable with widgets or `false(0)` otherwise
 */
-#define nk_tree_push(ctx, type, title, state) nk_tree_push_hashed(ctx, type, title, state, NK_FILE_LINE,nk_strlen(NK_FILE_LINE),__LINE__)
+#define nk_tree_push(ctx, type, title, state) nk_tree_push_hashed(ctx, type, title, state, nk_slicez(NK_FILE_LINE),__LINE__)
```

Comment:
I've just quickly tried to find all uses of this function and couldn't find anything convincing so I'm not sure it's necessary to break backwards compatibility here.

This seems to hold also for some other functions - e.g. nk_tree_image_push_id .

What was the motivation to change the API in such functions?

Reply:
I changed functions that took ptr+len already to take slices instead, for consistency. I can revert that as well, if we wonna keep complete backwards compatibility.

Reply:
Well, as much backwards compatibility as we can get that is

Reply:
> I changed functions that took ptr+len already to take slices instead, for consistency. I can revert that as well, if we wonna keep complete backwards compatibility.

Actually looking at it after the year and a half since you created this PR , with all the new "knowledge" (especially the trend to "duplicate" the string APIs) I think we could really try the "the more backwards compatible the better" approach. So I think we should keep str+len in existing functions :wink:.

But feel free to wait for others (@RobLoach etc.) to express themself before diving into the thousand line journey :wink:.

Reply:
> I think we could really try the "the more backwards compatible the better" approach. So I think we should keep str+len in existing functions wink.

I see. Maybe we should just drop the `nk_slice` and just take ptr+len as individual parameters. This would be consistent with what Nuklear already does and is what most C libraries in general does. I think we can also come up with better names for these functions than just an `*_s` prefix.

Reply:
>Maybe we should just drop the `nk_slice` and just take ptr+len as individual parameters.

Yep, why not. If it serves the intended purpose, then it's good :wink:!